test/binary.carp

```diff
@@ -0,0 +1,163 @@
+(use Binary)
+
+(load "Test.carp")
+(use Test)
+
+(deftest test
+  ;; int16 tests
+  (assert-equal test
+                (Uint16.from-long 2051l)
+                (unsafe-bytes->int16 (Binary.Order.LittleEndian) &[3b 8b])
+                "Unsafe little endian unsafe-bytes->int16 works as expected.")
+
+  (assert-equal test
+                (Uint16.from-long 776l)
+                (unsafe-bytes->int16 (Binary.Order.BigEndian) &[3b 8b])
+                "Unsafe big endian unsafe-bytes->int16 works as expected.")
+
+  (assert-equal test
+                &(Maybe.Just (Uint16.from-long 2051l))
+                &(bytes->int16 (Binary.Order.LittleEndian) &[3b 8b])
+                "Unsafe little endian bytes->int16 works as expected.")
+
+  (assert-equal test
+                &(Maybe.Just (Uint16.from-long 776l))
+                &(bytes->int16 (Binary.Order.BigEndian) &[3b 8b])
+                "Unsafe big endian bytes->int16 works as expected.")
+
+  (assert-equal test
+                &(Maybe.Nothing)
+                &(bytes->int16 (Binary.Order.LittleEndian) &[3b])
+                "bytes->int16 returns Nothing on insufficient data.")
+
+  (assert-equal test
+                &[(Uint16.from-long 2051l) (Uint16.from-long 776l)]
+                &(unsafe-bytes->int16-seq (Binary.Order.LittleEndian) &[3b 8b 8b 3b])
+                "Unsafe little endian bytes->int16-seq works as expected.")
+
+  (assert-equal test
+                &(Pair.init [(Uint16.from-long 776l) (Uint16.from-long 2051l)] 1)
+                &(bytes->int16-seq (Binary.Order.BigEndian) &[3b 8b 8b 3b 2b])
+                "Big endian bytes->int16-seq works as expected.")
+
+  ;; We unwrap the error here since it's simpler than defining equality over
+  ;; arrays of Uint values.
+  ;; TODO: Define equality for arrays of non-ref values.
+  (assert-equal test
+                &(Result.unsafe-from-error (the (Result (Array Uint16) Int) (Result.Error 1)))
```

Comment:
This is sadly a common necessity in tests atm. Maybe assertions for `Maybe` and `Result` might be a good idea as well, like `assert-just` or `assert-success`; that’s out of scope for this PR though.

Reply:
Thanks for the insight! Added #689 to track.